pkg/execution/executor/executor.go

```diff
@@ -173,6 +180,7 @@ type executor struct {
 	sm             state.Manager
 	al             coredata.ExecutionActionLoader
 	runtimeDrivers map[string]driver.Driver
+	failureHandler *func(context.Context, state.Identifier, state.State, state.DriverResponse) error
```

Comment:
This doesn't need to be a pointer, esp. when we convert this to a type :)

Reply:
@tonyhb Figured that this `WithX` pattern is for adding items that aren't required; wouldn't making this a value mean that it's always needed?

Just a meta question; I'm aware we always want it in this particular case!